packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -41,6 +42,13 @@ export const EMPTY_DETAIL_PANELS = Object.freeze(new Map<GridRowId, React.ReactN
 
 export type VirtualScroller = ReturnType<typeof useGridVirtualScroller>;
 
+const InfiniteLoadingTriggerElement = styled('div')({
+  position: 'sticky',
+  left: 0,
+  width: 0,
+  height: 0,
+});
+
```

Comment:
I don't love how the infinite loading logic is spilling in this file. Is there any other way we can separate the logic?

Instead of exposing `unstable_infiniteLoadingTriggerRef` on the API, maybe we could expose (a) function(s) to add the trigger element?

Reply:
The function could take as input the props, so the `props.rowsLoadingMode === 'client'` (below) would not need the `(rootProps as any)` cast that's required here.